ArduPlane/Plane.h

```diff
@@ -553,6 +554,9 @@ class Plane : public AP_HAL::HAL::Callbacks {
 
     // this controls throttle suppression in auto modes
     bool throttle_suppressed;
+	
+    // guiding roll from an external controller? (e.g., companion computer)
+    bool guiding_roll;
```

Comment:
I see you attempted to add guided_roll to auto_state, which was a good line of thinking. Lets make a new guided_state struct and put this into it